resources/files/windows/environment.bat

```diff
@@ -0,0 +1,35 @@
+@ECHO OFF
+REM This is the parent directory of the directory containing this script.
+SET PL_BASEDIR=%~dp0..
+REM Avoid the nasty \..\ littering the paths.
+SET PL_BASEDIR=%PL_BASEDIR:\bin\..=%
+
+REM Set a fact so we can easily source the environment.bat file in the future.
+SET FACTER_env_windows_installdir=%PL_BASEDIR%
+SET FACTERDIR=%PL_BASEDIR%
+
+REM Get the file name we were originally called as.  e.g. puppet.bat or puppet
+REM or facter.bat or facter.  ~n means: will return the file name only of
+SET SCRIPT_TEMP=%~n1
+REM Strip off the extension of the script name.  We need to do this to know
+REM what to pass to ruby -S
+SET SCRIPT_NAME=%SCRIPT_TEMP:.bat=%
+REM Shift off the original command name we we were called
+SHIFT
+
+SET PATH=%PL_BASEDIR%\bin;%PATH%

```

Comment:
Most of the PL_BASEDIR stuff seems unneeded (everything except `FACTER_env_windows_installdir`). I'm not sure about MCOLLECTIVED. Also, aren't SERVER_CONFIG and CLIENT_CONFIG the defaults for mco on Windows?